src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java

```diff
@@ -0,0 +1,157 @@
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.eq;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@WebMvcTest(controllers = UserCommonsController.class)
+public class UserCommonsControllerTests extends ControllerTestCase {
+
+  @MockBean
+  UserCommonsRepository userCommonsRepository;
+
+  @MockBean
+  UserRepository userRepository;
+
+  @MockBean
+  CommonsRepository commonsRepository;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @WithMockUser(roles = { "ADMIN" })
+  @Test
+  public void test_getUserCommonsById_exists_admin() throws Exception {
+  
+    UserCommons expectedUserCommons = UserCommons.dummyUserCommons(1);
+    when(userCommonsRepository.findByCommonsIdAndUserId(eq(1L),eq(1L))).thenReturn(Optional.of(expectedUserCommons));
+
+    MvcResult response = mockMvc.perform(get("/api/usercommons/?userId=1&commonsId=1"))
+        .andExpect(status().isOk()).andReturn();
+
+    verify(userCommonsRepository, times(1)).findByCommonsIdAndUserId(eq(1L),eq(1L));
+
+    String expectedJson = mapper.writeValueAsString(expectedUserCommons);
+    String responseString = response.getResponse().getContentAsString();
+
+    assertEquals(expectedJson, responseString);
+  }
+  
+  @WithMockUser(roles = { "ADMIN" })
+  @Test
+  public void test_getUserCommonsById_nonexists_admin() throws Exception {
+  
+    when(userCommonsRepository.findByCommonsIdAndUserId(eq(1L),eq(1L))).thenReturn(Optional.empty());
+
+    MvcResult response = mockMvc.perform(get("/api/usercommons/?userId=1&commonsId=1"))
+        .andExpect(status().is(404)).andReturn();
+
+    verify(userCommonsRepository, times(1)).findByCommonsIdAndUserId(eq(1L),eq(1L));
+    
+    String responseString = response.getResponse().getContentAsString();
+    String expected1 = "{\"message\":\"UserCommons with commonsId 1 and userId 1 not found\",\"type\":\"EntityNotFoundException\"}";
+    String expected2 = "{\"type\":\"EntityNotFoundException\",\"message\":\"UserCommons with commonsId 1 and userId 1 not found\"}";
+    
+    if(responseString.equals(expected1)) {
+      assertEquals(expected1, responseString);
+    }
+    else {
+      assertEquals(expected2, responseString);
+    }
```

Comment:
Why is this logic here? Is the ordering of the JSON properties inconsistent? It might be better to use the [`responseToJson`](https://github.com/ucsb-cs156-w22/team04-w22-6pm-HappyCows/blob/main/src/test/java/edu/ucsb/cs156/happiercows/ControllerTestCase.java#L34) method, as this workaround will not scale for larger JSON response bodies.

Reply:
`responseToJson` is used now instead